OptimizelySDK/Event/Builder/EventBuilder.cs

```diff
@@ -123,25 +125,23 @@ public void ResetParams()
 
         private Dictionary<string, object> GetImpressionParams(Experiment experiment, string variationId)
         {
-
             var impressionEvent = new Dictionary<string, object>();
 
             var decisions = new object[]
             {
                     new Dictionary<string, object>
                     {
-                        { Params.CAMPAIGN_ID,   experiment.LayerId },
-                        { Params.EXPERIMENT_ID, experiment.Id },
+                        { Params.CAMPAIGN_ID,   experiment?.LayerId },
+                        { Params.EXPERIMENT_ID, experiment?.Id ?? string.Empty },
```

Comment:
Is this ok to have empty string for EXPERIMENT_ID only?  CAMPAIGN_ID is also required in the event.

Reply:
I will fix in `UserEventFactory` and will prefer, this class shouldn't be  modified.